hashicorp/dev-portal
# feat: adds variants dropdown disclosure

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-ksadds-varaints-dropdown-disclosure-hashicorp.vercel.app) 🔎
- [Asana task](https://app.asana.com/0/1204333057896641/1204615849802262) 🎟️
- [Design file 🎨 ](https://www.figma.com/file/NAFjIPkbwjyY8jPl6T9nlP/Tutorial-Variants?type=design&node-id=238-39328&t=J7vcx4Y3jEZjOfta-0)

🚧 This PR is merging into an assembly 🚧 

## 🗒️ What

This PR adds a variant dropdown disclosure selector to the sidecar slot for the tutorial view. This component goes away at tablet, that is expected. We will add additional variants UI in future PRs.


<img width="935" alt="Screenshot 2023-05-16 at 3 58 28 PM" src="https://github.com/hashicorp/dev-portal/assets/36613477/96ab4bea-1923-4ef3-956c-3447f9c3921e">
<img width="311" alt="Screenshot 2023-05-16 at 4 19 47 PM" src="https://github.com/hashicorp/dev-portal/assets/36613477/c9bd49be-8aec-4a0b-991a-79e6809026dc">
<img width="334" alt="Screenshot 2023-05-16 at 4 19 37 PM" src="https://github.com/hashicorp/dev-portal/assets/36613477/ada0d5e3-1516-4f35-a11a-e24fe1953cc4">
<img width="374" alt="Screenshot 2023-05-16 at 4 34 07 PM" src="https://github.com/hashicorp/dev-portal/assets/36613477/8b061389-c31d-4bd5-a578-277adc643cb4">
<img width="351" alt="Screenshot 2023-05-16 at 4 34 10 PM" src="https://github.com/hashicorp/dev-portal/assets/36613477/629bab20-5be0-4283-bc30-298d454f9e69">



## 📸 Design Screenshots
<img width="183" alt="Screenshot 2023-05-16 at 3 59 56 PM" src="https://github.com/hashicorp/dev-portal/assets/36613477/340bc1a6-1607-4498-9aa7-712a18388dd7">
<img width="655" alt="Screenshot 2023-05-16 at 4 01 09 PM" src="https://github.com/hashicorp/dev-portal/assets/36613477/c9441108-b0c0-42a3-82fa-553ead8952ce">

## 🧪 Testing

- Visit a route with a [variant query param](https://dev-portal-git-ksadds-varaints-dropdown-disclosure-hashicorp.vercel.app/vault/tutorials/getting-started-ui/getting-started-policies-ui?variants=deployment:hcp) defined
- Open the dropdown, validate that the options show
- Use voice over to test again. Note the question on the activator aria label in the code comments.
- Use keyboard navigation to test, it should behave the exact same as all other dropdown disclosure components, no changes there.
- Add in more text to the variant name to validate the overflow state
- Validate that plain tutorial [paths without variants](https://dev-portal-git-ksadds-varaints-dropdown-disclosure-hashicorp.vercel.app/consul/tutorials/kubernetes/kubernetes-secure-agents) render as normal

## 💭 Anything else?
This PR uses `_tmp` data since the backend changes are still underway. I've tried to mark that explicitly with `@TODO` comments and `_tmp` prefixes.

🔴 _Note_ This PR hard codes the active variant option, so clicked a new variant option in the list will change the URL but the active option in the dropdown won't be reflected. That will happen when we hook up to real data. 

❓ I have [an open question ](https://github.com/hashicorp/dev-portal/pull/1928#issuecomment-1550495638)on how we handle the 'active' list item, whether it should be removed from the list like we do with version selector or we want to add an 'active' state.
